spec/support/fill_in_the_form_steps.rb

```diff
@@ -63,6 +63,16 @@ def and_has_given_their_name
     end
   end
 
+  def and_has_given_the_minimal_name_information
+    expect(page).to have_content(I18n.t("coronavirus_form.questions.name.title"))
+    within find(".govuk-main-wrapper") do
+      fill_in "first_name", with: "Test Please Ignore"
+      fill_in "middle_name", with: ""
```

Comment:
Can this line be removed from the test? It feels odd "filling in" something that the user hasn't touched.

Reply:
That's quite easy to test. I added `p params` as line 5 of `/app/controllers/coronavirus_form/name_controller.rb` and then ran `rspec  spec/features/filling_in_the_form/fill_in_the_form_spec.rb:30` with the code is. The `params` output was:
```ruby
<ActionController::Parameters {"first_name"=>"Test Please Ignore", "middle_name"=>"", "last_name"=>"Test Please Ignore", "controller"=>"coronavirus_form/name", "action"=>"submit"} permitted: false>
```
Then I commented out line 70 and the output looked like this:
```ruby
<ActionController::Parameters {"first_name"=>"Test Please Ignore", "middle_name"=>"", "last_name"=>"Test Please Ignore", "controller"=>"coronavirus_form/name", "action"=>"submit"} permitted: false>
```
As you can see, exactly the same.

These test are using capybara, so the actions are being carried out by a headless browser - therefore the page will be rendered headlessly, and the submit button clicked. If nothing is entered in a text field it will behave the same as with a normal browser - an empty string is submitted.

So - long winded - but in conclusion these `fill_in "<xxxx>", with ""` are redundant.